commit bc83add2e2c5c0a1e05082cac76cfba5411a1a3c (refs/changes/49/4849/5)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2018-09-20T20:00:46+00:00 (1 year, 1 month ago)
    
    MANTA-3950 muskie_deleted_bytes metric should look at the creator of an object instead of the caller
    Reviewed by: Jared Morrow <jm@joyent.com>
    Approved by: Jared Morrow <jm@joyent.com>

diff --git a/lib/audit.js b/lib/audit.js
index 59a5d27..795bfea 100644
--- a/lib/audit.js
+++ b/lib/audit.js
@@ -300,26 +300,23 @@ function auditLogger(options) {
         }
 
         if (op === 'DELETE') {
+            var md = req.metadata;
+            var owner = md.creator || md.owner;
+
             /*
-             * Delete metrics for snaplink-disabled accounts provide a
-             * meaningful measure of short-term storage reclamation in
-             * deployments with garbage-collectors.
-             *
-             * We avoid tracking these metrics for accounts that are not
-             * snaplink-disabled because the storage for objects owned by such
-             * accounts may still be occupied for long periods of time after
-             * the corresponding delete API calls complete.
+             * Count bytes for which DELETE API requests have been completed. If
+             * the owner of the underlying data has snaplinks disabled, then
+             * the deleted object was processed by the accelerated deletion
+             * pipeline.
              */
-            common.checkAccountSnaplinksEnabled(req, req.caller.account.uuid,
+            common.checkAccountSnaplinksEnabled(req, owner,
                 function (enabled) {
-                if (enabled) {
-                    return;
-                }
-                var md = req.metadata;
                 if (md.type === 'object' && md.contentLength > 0) {
                     var storage = md.contentLength * md.sharks.length;
-                    labels.login = req.caller.account.login ||
-                        req.caller.account.uuid;
+                    labels = {
+                        accelerated_gc: !enabled,
+                        owner: owner
+                    };
                     deleted_data_counter.add(storage, labels);
                 }
             });
diff --git a/lib/server.js b/lib/server.js
index a7a3780..c23229d 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -147,7 +147,7 @@ function createServer(options, clients, name) {
     });
     options.collector.counter({
         name: common.METRIC_DELETED_DATA_COUNTER,
-        help: 'count of bytes deleted by snaplink-disabled clients'
+        help: 'count of deleted object bytes'
     });
 
     var _timeout = parseInt((process.env.SOCKET_TIMEOUT || 120), 10) * 1000;
